### ISSUE TITLE
No kind logs when a job times out after 2 hours

### ISSUE THREAD

**Issue by maelvls** [jetstack/testing#650]
When the Prow runner times out (2 hours), two unwanted behaviors occur:

1. The container logs (e.g., pebble) aren't uploaded to the "artifacts" because `make kind-logs` does not seem to be called,
2. The Prow UI doesn't show the test cases that passed and failed because Ginkgo does not produce any XML output.

Initially, I thought I had forgotten to set a trap for `SIGINT` that gets sent when the timeout is reached, because Ginkgo seemed to keep going without stopping. But in reality, these two symptoms are due to the [`runner` script](https://github.com/jetstack/testing/blob/cb6f068d60b4819fef0f9521a9c4f849d4d97928/images/bazelbuild/runner) not retransmitting SIGINT it to its children when it receives SIGINT. This is because bash ignores any signals while a child process is executing with `command "$@"`.

Here is what a Prow job looks like:

```yaml
command: /tools/entrypoint
args:
  - runner
  - bash
  - -c
  - |
    apt-get install jq -y >/dev/null
    make -j vendor-go e2e-ci K8S_VERSION=1.23
```

Let us mimick what `runner` does:

```sh
$ bash -c 'echo $$; command bash -c "trap \"echo CLEANING UP\" EXIT; sleep 1000"' &
# <----------------------------------------------------------------------------->
#                 Mimicks the command "runner".
#                  <------------------------------------------------------------>
#                  Mimicks the command "bash -c apt-get..."
#                  that is run inside the "runner" script with
#                  the line "command $@".
```

It should output the PID of the inner bash process:

```text
2896855
```

Now, type:

```sh
kill -s INT 2896855
```

Nothing happens, the bash script should still be running. The outer bash script has not passed the SIGINT signal down to its children.

But with SIGTERM, it seems to work:

```console
$ kill -s TERM 2896855
CLEANING UP
[1]  + 2896855 terminated  bash -c
```

After fixing this issue, I suggest reducing the overall timeout from 2 hours to 40 minutes.

**Comment by munnerz** [comment]
This takes me back a while... https://github.com/jetstack-experimental/php-fpm-autoreload/blob/963b7fb64587ec4063bfc5aa0ff89e65578927e8/entrypoint.sh#L23-L29

One option is to capture the signal being sent to the bash script and remap it onto the relevant process :)

**Comment by maelvls** [comment]
Ohh thank you!! Thank to your idea, the following works as expected:

```console
$ bash -c 'echo $$; command bash -c "trap \"echo CLEANING UP\" EXIT; sleep 1000" & trap "kill INT $! 2>/dev/null" SIGINT; wait $!' &
2896855
```

then 

```console
$ kill INT 2896855
CLEANING UP
```

**Comment by maelvls** [comment]
Today ([slack thread](https://jetstack.slack.com/archives/C63VC8FAT/p1661173462458039?thread_ts=1661160866.230979&cid=C63VC8FAT)), we have hit another occurrence of the symptom (1): because `make kind-logs` isn't being run, the logs of the pebble pod aren't uploaded, preventing us from debugging an issue with Pebble.

**Comment by jahrlin** [comment]
A workaround for problem 1 is to use the Logs Explorer in GCP, and query with the build ID, e.g.

```
labels."k8s-pod/prow_k8s_io/build-id"="1560906990065029120"
resource.labels.container_name="test"
```

This will show all the output from the containers of the container in which ginkgo runs, which includes test results - but hard to parse for humans

e.g. https://cloudlogging.app.goo.gl/gFp2UV6Hqd9R7Q3YA
